BootloaderCommonPkg/Library/SmbiosInitLib/SmbiosInitLib.c

```diff
@@ -244,11 +245,26 @@ GetSmbiosString (
 
   for (Index = 0; Index < PcdGet16 (PcdSmbiosStringsCnt); ++Index) {
     if (Type == SmbiosStrings[Index].Type && StringId == SmbiosStrings[Index].Idx) {
-      return SmbiosStrings[Index].String;
+      if (SmbiosStrings[Index].String != NULL) {
+        if (AsciiStrLen (SmbiosStrings[Index].String) != 0) {
+          StringFound = TRUE;
+          break;
+        } else {
+          StringFound = FALSE;
+          break;
+        }
+      } else {
+        StringFound = FALSE;
+        break;
+      }
```

Comment:
The patch and logic would be very simple if using below code to replace "return SmbiosStrings[Index].String;"
if ((SmbiosStrings[Index].String != NULL) && (AsciiStrLen (SmbiosStrings[Index].String) != 0)) {
    return AsciiStrLen (SmbiosStrings[Index].String;
}
break;